src/postprocess/stresses.jl

```diff
@@ -0,0 +1,37 @@
+using ForwardDiff
+"""
+Compute the stresses (= 1/Vol dE/d(M*lattice), taken at M=I) of an obtained SCF solution.
+"""
+@timing function compute_stresses(basis::PlaneWaveBasis, ψ, occ; kwargs...)
+    # TODO optimize by only computing derivatives wrt 6 independent parameters
+    # TODO this is a bit ugly (couples strongly to the structures) because we
+    # don't have a good way to say "make the same model/pwbasis except for this"
+
+    # compute the Hellmann-Feynman energy (with fixed ψ/occ/ρ)
+    function HF_energy(lattice)
+        T = eltype(lattice)
+        # mimic Model constructor
+        model = basis.model
+        new_model = Model(lattice;
+                          model.n_electrons,
+                          model.atoms,
+                          magnetic_moments=[], # not used because we give symmetries explicitly
+                          terms=model.term_types,
+                          model.temperature,
+                          model.smearing,
+                          model.spin_polarization,
+                          model.symmetries)
+        new_basis = PlaneWaveBasis(new_model, basis.Ecut,
+                                   basis.kcoords_global, basis.ksymops_global,
+                                   basis.fft_size, basis.variational, basis.symmetries,
+                                   basis.kgrid, basis.kshift, basis.comm_kpts)
+        ρ = DFTK.compute_density(new_basis, ψ, occ)
+        energies, _ = energy_hamiltonian(new_basis, ψ, occ; ρ=ρ)
+        energies.total
+    end
+    ForwardDiff.gradient(M -> HF_energy((I+M) * basis.model.lattice),
+                         zeros(eltype(basis), 3, 3)) / det(basis.model.lattice)
+end
+function compute_stresses(scfres)
+    compute_stresses(scfres.basis, scfres.ψ, scfres.occupation)
+end
```

Comment:
I guess one should also have a `stresses_cart` for consistency?

Reply:
I'm not super confortable with stress definitions so I'd rather have just one that matches what's found in the literature. Forces are more intuitive

Reply:
I see, but forces in the literature are usually cartesian as well. Perhaps we should only support cartesian forces? Or we wait until we understand more and decide then.

Reply:
Well OK, cartesian stresses might make sense. Forces in reduced coordinates definitely makes sense, it's what you want to do for geometry optimization for instance. For stresses I'm not sure what to do about the division by volume, so let's leave it as is for now.